functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -79,235 +88,341 @@ Run the selection made by the user on the Sql Server instance specified.
+						}
+
+						try {
+							$result = $server.Query($scriptpart.Text)
+							Write-Message -Level Verbose -Message "Processed $($scriptpart.QueryName) on $instance"
+							if (!$result) {
+								[void]$QueryDetail.Add( [pscustomobject]@{
+										ComputerName     = $server.NetName
+										InstanceName     = $server.ServiceName
+										SqlInstance      = $server.DomainInstanceName
+										Number           = $scriptpart.QueryNr
+										Name             = $scriptpart.QueryName
+										Description      = $scriptpart.Description
+										DatabaseSpecific = $scriptpart.DBSpecific
+										DatabaseName     = $null
+										Notes            = "Empty Result for this Query"
+										Result           = $null
+									})
+								Write-Message -Level Verbose -Message ("Empty result for Query {0} - {1} - {2}" -f $scriptpart.QueryNr, $scriptpart.QueryName, $scriptpart.Description)
+							}
+						}
+						catch {
+							Write-Message -Level Verbose -Message ('Some error has occured on Server: {0} - Script: {1}, result unavailable' -f $instance, $scriptpart.QueryName) -Target $instance -ErrorRecord $_
+						}
+						if ($result) {
+							[void]$QueryDetail.Add( [pscustomobject]@{
+									ComputerName     = $server.NetName
+									InstanceName     = $server.ServiceName
+									SqlInstance      = $server.DomainInstanceName
+									Number           = $scriptpart.QueryNr
+									Name             = $scriptpart.QueryName
+									Description      = $scriptpart.Description
+									DatabaseSpecific = $scriptpart.DBSpecific
+									DatabaseName     = $null
+									Notes            = $null
+									Result           = $result | Select-Object * -ExcludeProperty RowError, RowState, Table, ItemArray, HasErrors
+								}
+							)
+						}
+					}
+					else {
+						# if running WhatIf, then return the queries that would be run as an object, not just whatif output
+						[void]$QueryDetail.Add(
+							[pscustomobject]@{
+								ComputerName     = $server.NetName
+								InstanceName     = $server.ServiceName
+								SqlInstance      = $server.DomainInstanceName
+								Number           = $scriptpart.QueryNr
+								Name             = $scriptpart.QueryName
+								Description      = $scriptpart.Description
+								DatabaseSpecific = $scriptpart.DBSpecific
+								DatabaseName     = $null
+								Notes            = "WhatIf - Bypassed Execution"
+								Result           = $null
+							})
+					}
+
+				}
+				elseif ($scriptpart.DBSpecific -and !$instanceOnly) {
+					foreach ($currentdb in $databases) {
+						$dbname = $currentdb.name
+
+						if ($ExportQueries) {
+							[io.directory]::CreateDirectory($OutputPath)>$null
+							$FileName = Remove-InvalidFileNameChars ('{0}-{1}-{2}.sql' -f $server.DomainInstanceName, $dbname, $Scriptpart.QueryName)
+							$FullName = ([io.path]::combine($OutputPath, $FileName))
+							write-verbose "Creating file: $FullName" -Verbose
```

Comment:
watch out for write-verbose

Reply:
You can see the results directly on appveyor (see "All checks have failed" --> "Appveyor build failed" right on this page)

Reply:
as for buffering, you can see other commands *usually* just return pscustomobject to the pipeline as soon as they come, without "witholding" them until the end of the function execution. In fact, such usage is discouraged but allowed **if needed**. Is there a need to return the output only in the end block ?

Reply:
Ok. I see. I normally do that to try and be more explicit about what is being returned as I cast the return object. If that is not desired it should be an easy change. Thanks for the feedback